pkg/operation/botanist/component/extensions/dns/dnsentry.go

```diff
@@ -30,64 +28,78 @@ import (
 	"github.com/sirupsen/logrus"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 )
 
 // EntryValues contains the values used to create a DNSEntry
 type EntryValues struct {
-	Name    string   `json:"name,omitempty"`
-	DNSName string   `json:"dnsName,omitempty"`
-	Targets []string `json:"targets,omitempty"`
-	OwnerID string   `json:"ownerID,omitempty"`
-	TTL     int64    `json:"ttl,omitempty"`
+	Name    string
+	DNSName string
+	Targets []string
+	OwnerID string
+	TTL     int64
 }
 
-// NewDNSEntry creates a new instance of DeployWaiter for a specific DNS entry.
+// NewEntry creates a new instance of DeployWaiter for a specific DNS emptyEntry.
 // <waiter> is optional and it's defaulted to github.com/gardener/gardener/pkg/utils/retry.DefaultOps()
-func NewDNSEntry(
-	values *EntryValues,
-	shootNamespace string,
-	applier kubernetes.ChartApplier,
-	chartsRootPath string,
+func NewEntry(
 	logger logrus.FieldLogger,
```

Comment:
nit: if you reorder the params anyway do you want to do it alphabetically?

Reply:
No, we often have logger first, client next, etc. There is no strict guideline (and I don't think we need one), I just aligned it with the majority of the other code.